compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2451,6 +2451,17 @@ class SynchronizedCallOnBoxedClass(stat: tpd.Tree)(using Context)
         |you intended."""
 }
 
+class ExtensionNullifiedByMember(method: Symbol, target: Symbol)(using Context)
+  extends Message(ExtensionNullifiedByMemberID):
+  def kind = MessageKind.PotentialIssue
+  def msg(using Context) =
+    i"""Extension method ${hl(method.name.toString)} will never be selected
+       |because ${hl(target.name.toString)} already has a member with the same name."""
```

Comment:
Maybe we can be a bit more precise without getting into too many details:
```suggestion
       |because ${hl(target.name.toString)} already has a member with the same name and compatible parameter types."""
```